gobblin-runtime/src/main/java/org/apache/gobblin/runtime/AbstractJobLauncher.java

```diff
@@ -598,7 +598,7 @@ public void apply(JobListener jobListener, JobContext jobContext)
               }
             });
 
-            if (jobState.getState() == JobState.RunningState.FAILED) {
+            if (jobState.getState() == JobState.RunningState.FAILED || jobState.getState() == JobState.RunningState.CANCELLED) {
```

Comment:
Is there another bug in "else" branch below? It creates a "JOB_SUCCEEDED" event, but the listener is calling "onJobFailure" event. Also the same logic is a coupe of blocks above.

Reply:
It's been like that for a long time, I'm not sure if that was originally intentional or not. But there is no `onJobSuccess` API, so I'm not sure what would replace it. Also would have to be careful changing something here, since changing logic could affect many gobblin jobs.